### PR TITLE
Fix-up numbering and add test dependencies

### PR DESCRIPTION
**Pull request by whatnick**
### Reason for this pull request

Updates to README for first time users/builders

### Proposed changes

Update readme for conda build steps to include test requirments.

 - [ ] Closes #xxxx
 - [ ] Tests added / passed
 - [ ] Fully documented, including `docs/about/whats_new.rst` for all changes

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -61,14 +61,18 @@ Developer setup
 
    pre-commit install
 
-4. Run unit tests + PyLint
-
+5. Run unit tests + PyLint
+
```
Comment (Kirill888) [inline README.rst]: minor nitpick: trailing white space on line 65

File: README.rst [hunk 2]
```diff
@@ -61,14 +61,18 @@ Developer setup
...
 
-4. Run unit tests + PyLint
-
+5. Run unit tests + PyLint
+   
    ``./check-code.sh``
 
    (this script approximates what is run by Travis. You can
-   alternatively run ``pytest`` yourself)
+   alternatively run ``pytest`` yourself). Some test dependencies may need to be install, attempt to install these using:
```
Comment (Kirill888) [inline README.rst]: "may need to be install**ed**"

**Comment by whatnick** [comment]
Addressed comments.

**Review by Kirill888 — APPROVED** [review]
:+1:

**Pull request merged**